### PR TITLE
Dan/rollouts from actions

### PR DESCRIPTION
**Pull request by danielsdeleo**
### :nut_and_bolt: Description: What code changed, and why?

Create rollouts from Chef Server actions messages alongside the external metadata API. The preferred method for creating rollouts will be the client-side metadata collector, which provides more useful data about the code being rolled out, but we also wish to create rollouts from Chef Server action messages for the following cases:
* Ensure that we create at least a minimal record of new code rollouts for cases where the metadata collector isn't properly configured, the user takes out of band action on a system where there metadata collector isn't configured, etc.
* Give a preview of the feature to users who are deciding whether to adopt.

This behavior is implemented by forwarding policy update actions from the ingest service to the config management service, which creates rollout records for the incoming actions if they do not exist. More specifically:
* Add a bind at the hab and deployment service level so ingest can make a connection to config management
* Add a publisher in the ingest service actions pipeline to send the messages to config management
* Make an API endpoint for config management to accept policy update action data
* Insert a rollout in the rollouts table when a policy update action is received. In case of a conflict (i.e., we already have a record of this rollout), do nothing.
* Modify the API that the metadata collector uses to create rollouts, so that it does an upsert. The "happy path" will be that we get a metadata API call and an actions message for each new rollout, and we cannot control which one comes first. If the actions message comes first, we will have a "bare bones" record of the rollout in the database, and we need to add the extra metadata to that row. We use postgres' built-in upsert to help avoid concurrency issues.

### :chains: Related Resources

#3987 #2873

### :athletic_shoe: How to Build and Test the Change

* `build components/automate-deployment/ && build components/ingest-service/ && build components/config-mgmt-service/`
* I think the dev channel is behind master because of a pipeline issue. Because we changed deployment service and rebuilt from master, this means you need to `build components/event-feed-service/` :| 
* `start_all_services` and `start_chef_server` and `bootstrap_chef_user_data`
* `hab pkg install chef/chef-dk -bf` to get the `chef` CLI in your PATH
* Get a policyfile installed so you can push it to a chef server (you'll have to accept the Chef license)`pushd /tmp/ && git clone https://github.com/danielsdeleo/policyfile-jenkins-demo.git && cd policyfile-jenkins-demo && chef install`
* `chef push -c /tmp/.client.rb foo` to upload the policy to the Chef Server
* `config_mgmt_list_rollouts` to see the rollouts

### CODE REVIEW

File: components/config-mgmt-service/grpcserver/actions_handler.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	"google.golang.org/grpc/status"
+
+	"github.com/chef/automate/api/interservice/cfgmgmt/request"
+	"github.com/chef/automate/api/interservice/cfgmgmt/response"
+	"github.com/chef/automate/components/config-mgmt-service/backend/postgres"
+)
+
+func (s *CfgMgmtServer) HandlePolicyUpdateAction(ctx context.Context, req *request.PolicyUpdateAction) (*response.PolicyUpdateAction, error) {
+	// FIXME remove this or make it debug w/e
+	log.Infof("got HandlePolicyUpdateAction call %+v", *req)
```
Comment (lancewf) [inline components/config-mgmt-service/grpcserver/actions_handler.go:19]: You have a note to remove this line

File: components/ingest-service/pipeline/publisher/config_mgmt_publisher.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	start := time.Now()
+
+	req := &request.PolicyUpdateAction{
+		PolicyName:         msg.InternalChefAction.EntityName,
+		PolicyGroup:        msg.InternalChefAction.ParentName,
+		PolicyRevisionId:   msg.InternalChefAction.RevisionId,
+		ChefServerFqdn:     msg.InternalChefAction.ServiceHostname,
+		ChefServerOrgname:  msg.InternalChefAction.OrganizationName,
+		ChefServerUsername: msg.InternalChefAction.RequestorName,
+		PolicyfileContent:  msg.InternalChefAction.Data,
```
Comment (lancewf) [inline components/ingest-service/pipeline/publisher/config_mgmt_publisher.go:58]: I will just parse out all the cookbooks from the "Data" field and add a new field to the PolicyUpdateAction.

File: components/ingest-service/pipeline/publisher/config_mgmt_publisher.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+	logCtx := log.WithFields(log.Fields{
+		"message_id":  msg.ID,
+		"buffer_size": len(out),
+		"dur":         time.Since(start),
+	})
+
+	if err != nil {
+		logCtx.WithError(err).Error("Failed to publish actions message to config management service")
+		return
```
Comment (lancewf) [inline components/ingest-service/pipeline/publisher/config_mgmt_publisher.go:74]: If it fails to publish the action to the config-mgmt-service we should fail the pipeline with `msg.FinishProcessing(err)`

Comment (danielsdeleo) [reply components/ingest-service/pipeline/publisher/config_mgmt_publisher.go:74]: 👍 will do, I wasn't sure if we wanted to try a best effort to pass it to the next publisher

File: components/config-mgmt-service/grpcserver/actions_handler.go (hunk 1 above)
Comment (danielsdeleo) [reply components/config-mgmt-service/grpcserver/actions_handler.go:19]: haha, I looked for TODOs before I pushed

**Pull request merged**